pkg/geo/geomfn/generate_points.go

```diff
@@ -0,0 +1,217 @@
+	}
+	if area == 0.0 {
+		return nil, errors.New("zero area input Polygon")
+	}
+	bbox := g.CartesianBoundingBox()
+	bboxWidth := bbox.HiX - bbox.LoX
+	bboxHeight := bbox.HiY - bbox.LoY
+	bboxArea := bboxHeight * bboxWidth
+	// Gross up our test set a bit to increase odds of getting coverage in one pass.
+	sampleNPoints := float64(nPoints) * bboxArea / area
+
+	sampleSqrt := math.Round(math.Sqrt(sampleNPoints))
+	if sampleSqrt == 0 {
+		sampleSqrt = 1
+	}
+	var sampleHeight, sampleWidth int
+	var sampleCellSize float64
+	// Calculate the grids we're going to randomize within.
+	if bboxWidth > bboxHeight {
+		sampleWidth = int(sampleSqrt)
+		sampleHeight = int(math.Ceil(sampleNPoints / float64(sampleWidth)))
+		sampleCellSize = bboxWidth / float64(sampleWidth)
+	} else {
+		sampleHeight = int(sampleSqrt)
+		sampleWidth = int(math.Ceil(sampleNPoints / float64(sampleHeight)))
+		sampleCellSize = bboxHeight / float64(sampleHeight)
+	}
+	// Prepare the polygon for fast true/false testing.
+	gPrep, err := geos.PrepareGeometry(g.EWKB())
+	if err != nil {
+		return nil, errors.Newf("could not prepare geometry: %v", err)
+	}
+	res, err := func() (*geom.MultiPoint, error) {
+		// Generate a slice of points - for every cell on a grid store coordinates.
+		n := sampleHeight * sampleWidth
+		cells := make([]geom.Coord, n)
+		for i := 0; i < sampleWidth; i++ {
+			for j := 0; j < sampleHeight; j++ {
+				cells[i*sampleHeight+j] = geom.Coord{float64(i), float64(j)}
+			}
+		}
+		// Shuffle the points. Without shuffling, the generated point will
+		// always be adjacent to the previous one (in terms of grid cells).
+		if n > 1 {
+			rng.Shuffle(n, func(i int, j int) {
+				temp := cells[j]
+				cells[j] = cells[i]
+				cells[i] = temp
+			})
+		}
+		results := geom.NewMultiPoint(geom.XY)
+		// Generate points and test them.
+		nPointsGenerated := 0
+		iterations := 0
+		for nPointsGenerated < nPoints {
+			iterations++
+			for _, cell := range cells {
+				y := bbox.LoY + cell.X()*sampleCellSize
+				x := bbox.LoX + cell.Y()*sampleCellSize
+				x += rng.Float64() * sampleCellSize
+				y += rng.Float64() * sampleCellSize
+				if x > bbox.HiX || y > bbox.HiY {
+					continue
+				}
+				gpt, err := geo.MakeGeometryFromPointCoords(x, y)
+				if err != nil {
+					return nil, errors.Newf("could not create geometry Point: %v", err)
+				}
+				intersects, err := geos.PreparedIntersects(gPrep, gpt.EWKB())
+				if err != nil {
+					return nil, errors.Newf("could not check prepared intersection: %v", err)
+				}
+				if intersects {
+					nPointsGenerated++
+					p := geom.NewPointFlat(geom.XY, []float64{x, y})
+					srid := g.SRID()
+					p.SetSRID(int(srid))
+					err = results.Push(p)
+					if err != nil {
+						return nil, errors.Newf("could not add point to the results: %v", err)
+					}
+					if nPointsGenerated == nPoints {
+						break
+					}
+				}
+			}
+			if nPointsGenerated == nPoints || iterations > 100 {
```

Comment:
I was not sure, so used the PostGIS value. I think it's just an arbitrary number to avoid going through too many point generations if they are not covered by the geometry.
Yes, fixed.